packages/arb-avm-cpp/data_storage/src/checkpointedmachine.cpp

```diff
@@ -193,72 +180,45 @@ void CheckpointedMachine::saveAssertion(const Assertion& assertion) {
                                  status.ToString());
     }
 
-    pending_checkpoint.step_count += +assertion.stepCount;
-    pending_checkpoint.messages_read_count += assertion.inbox_messages_consumed;
-    pending_checkpoint.logs_output += assertion.logs.size();
-    pending_checkpoint.messages_output += assertion.outMessages.size();
     pending_checkpoint.arb_gas_used += assertion.gasCount;
+    pending_checkpoint.message_sequence_number_processed =
+        first_message_sequence_number + assertion.inbox_messages_consumed - 1;
+    pending_checkpoint.step_count += assertion.stepCount;
+    pending_checkpoint.send_count += assertion.sends.size();
+    pending_checkpoint.log_count += assertion.logs.size();
 }
 
-uint64_t CheckpointedMachine::reorgToMessageOrBefore(
-    const uint64_t& message_number) {
+uint256_t CheckpointedMachine::reorgToMessageOrBefore(
+    const uint256_t& message_sequence_number) {
     auto tx = Transaction::makeTransaction(data_storage);
 
-    auto result = getCheckpointAtOrBeforeMessage(message_number);
+    auto result = getCheckpointAtOrBeforeMessage(message_sequence_number);
     if (!result.status.ok()) {
         throw std::runtime_error("error getting checkpoint for reorg: " +
                                  result.status.ToString());
     }
 
-    ValueCache value_cache{};
+    // Truncate sends and logs
     pending_checkpoint = result.data;
     static_cast<AggregatorStore>(data_storage)
```

Comment:
This will presumably be in the new core db instead of the aggregator store

Reply:
moved to its own family column